libs/sdk-core/src/greenlight/error.rs

```diff
@@ -186,7 +186,11 @@ impl From<tonic::Status> for NodeError {
                 JsonRpcErrCode::InvoiceNoDescription => Self::InvoiceNoDescription(status.into()),
                 JsonRpcErrCode::InvoicePreimageAlreadyExists => {
                     Self::InvoicePreimageAlreadyExists(status.into())
-                }
+                },
+                JsonRpcErrCode::OfferExpired => Self::OfferExpired(status.into()),
+                JsonRpcErrCode::OfferBadInvreqReply => Self::OfferReplyError(status.into()),
+                JsonRpcErrCode::OfferRouteNotFound => Self::RouteNotFound(status.into()),
```

Comment:
Thinking out loud. I think it may be useful to distinguish between 'payment' route not found and 'messaging' route not found. RouteNotFound is such a general error that pops up too many times. @dangeross what do you think?